src/credentialstore/credentialstore.ts

```diff
@@ -26,44 +33,50 @@ export class CredentialStore implements ICredentialStore {
 	 * @param {string} credentialId the ID uniquely identifying this credential
 	 * @returns {Promise<Credential>} Promise that resolved to the credential, or undefined if not found
 	 */
-	public readCredential(credentialId: string): Promise<Contracts.Credential> {
+	public async readCredential(credentialId: string): Promise<Contracts.Credential> {
 		let self = this;
 		let cred: Contracts.Credential = new Contracts.Credential();
 		cred.credentialId = credentialId;
-		return new Promise<Contracts.Credential>((resolve, reject) => {
-			self._client
-				.sendRequest(Contracts.ReadCredentialRequest.type, cred)
-				.then(returnedCred => {
-					resolve(returnedCred);
-				}, err => reject(err));
-		});
+		try {
+			const returnedCred = await self._client.sendRequest(Contracts.ReadCredentialRequest.type, cred)
+			return returnedCred;
+		} catch (error) {
```

Comment:
You can just get rid of the try/catch if you're immediately rethrowing the error. This is doing nothing right now.